llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -940,6 +940,18 @@ static void simplifyRecipe(VPRecipeBase &R, VPTypeAnalysis &TypeInfo) {
 #endif
   }
 
+  VPValue *B;
+  VPValue *C;
+  VPValue *D;
+  // Simplify (X && Y) || (X && !Y) -> X.
+  if (match(&R,
```

Comment:
More general and modular: first simplify (A && B) || (C && D) where A == C into A && (B || D). Also for A == D, B == C, B == D. Followed by simplifying (X || !X).

Reply:
ok, worth leaving behind a TODO?

Reply:
Done, thanks!